### PR TITLE
Task/gamification initial code setup

### PR DESCRIPTION
**Pull request by priyankalekkala**
getting hours from approved_applicant table

### CODE REVIEW

File: DCode.Data/DbContexts/ApprovedApplicantDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+   public class ApprovedApplicantDbContext: DbContext
+    {
+        public ApprovedApplicantDbContext()
+            : base(Constants.DCodeEntities)
+        {
+            this.Configuration.LazyLoadingEnabled = false;
+        }
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            throw new UnintentionalCodeFirstException();
```
Comment (shivam-r) [inline DCode.Data/DbContexts/ApprovedApplicantDbContext.cs:16]: Hey @priyankalekkala ,
Please check this I am not confident if we should throw unhandled exception here.

File: DCode.Services/Common/CommonService.cs [hunk 2]
```diff
@@ -502,6 +506,13 @@ public IEnumerable<Offering> GetOfferings()
             return _offeringModelFactory.CreateModelList<Offering>(offerings);
         }
 
+        public decimal? GetApprovedApplicantHours()
+        {
+            var applicants = _approvedApplicantRepository.GetApprovedApplicants();
+            var hoursWorked = _approvedApplicantModelFactory.CreateModelList<ApprovedApplicant>(applicants).Where(x => x.APPLICANT_ID == 649);
```
Comment (shivam-r) [inline DCode.Services/Common/CommonService.cs]: there is HardCoding here 
Is this intentional?

**Review by shivam-r — DISMISSED** [review]

File: DCode.Data/MetadataRepository/IApprovedApplicantRepository.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using DCode.Data.DbContexts;
+using System.Collections.Generic;
+
+namespace DCode.Data.MetadataRepository
+{
+    public interface IApprovedApplicantRepository
+    {
+        IEnumerable<approvedapplicant> GetApprovedApplicants();
```
Comment (ritwiksen) [inline DCode.Data/MetadataRepository/IApprovedApplicantRepository.cs:8]: Can we not instead create a method which fetches the approved applicant only on the basis of the `Applicant_ID`? We need not fetch the data of the whole table into the service layer.

File: DCode.Services/Common/CommonService.cs (hunk 2 above)
Comment (ritwiksen) [reply DCode.Services/Common/CommonService.cs]: @priyankalekkala :: Should we not get the total hours of the current user alone? Please refer to my comment as well for `IApprovedApplicantRepository `.

**Review by ritwiksen — CHANGES_REQUESTED** [review]

File: DCode.Data/MetadataRepository/IApprovedApplicantRepository.cs (hunk 3 above)
Comment (ritwiksen) [reply DCode.Data/MetadataRepository/IApprovedApplicantRepository.cs:8]: This is addressed.

**Review by ritwiksen — APPROVED** [review]

**Pull request closed without merging**